lib/openshift4-monitoring-alert-patching.libsonnet

```
@@ -108,6 +108,9 @@ local patchRule(rule, patches={}, patchName=true) =
         // mark alert as belonging to component instance in whose context the
         // function is called.
         syn_component: inv.parameters._instance,
+        // mark alert as belonging to the team in whose context the
+        // function is called.
+        [if std.objectHas(inv.parameters, 'syn') then 'syn_team']: std.get(inv.parameters.syn, 'owner', ''),
```

As described in the internal ticket, this should also take into account `syn.teams`. Something like

```jsonnet
local syn_team =
  local instance = inv.parameters._instance;
  local syn = std.get(inv.parameters, 'syn', { owner: '', teams: {} });
  local team_instances = [
    {
      team: tn,
      instances: syn.teams[tn].instances,
    }
    for tn in std.objectFields(syn.teams)
  ];
  local team = std.foldl(
    function(o, ti)
      if std.member(ti.instances, instance) then
        o + [ ti.team ]
      else
        o,
    team_instances,
    []
  );
  if std.length(team) > 1 then
    error "Multiple owners for instance '%s': %s" % [ instance, team ]
  else if std.length(team) == 1 then
    team[0]
  else
    syn.owner;
```

And just use the new local variable in the label set:

```suggestion
        syn_team: syn_team,
```

---

Looks like I interpreted this wrongly.